schemes/check_energy/check_energy_chng.F90

```diff
@@ -0,0 +1,414 @@
+      call get_hydrostatic_energy(                               &
+          tracer             = q(1:ncol,1:pver,1:pcnst),         & ! moist mixing ratios
+          moist_mixing_ratio = .true.,                           &
+          pdel_in            = pdel           (1:ncol,1:pver),   &
+          cp_or_cv           = cp_or_cv_dycore(1:ncol,1:pver),   &
+          U                  = u              (1:ncol,1:pver),   &
+          V                  = v              (1:ncol,1:pver),   &
+          T                  = T              (1:ncol,1:pver),   & ! enthalpy-scaled temperature for energy consistency
+          vcoord             = energy_formula_dycore,            & ! energy formula for dycore
+          ptop               = pintdry        (1:ncol,1),        &
+          phis               = phis           (1:ncol),          &
+          z_mid              = z_ini          (1:ncol,:),        & ! unique for MPAS
+          te                 = te_ini_dyn     (1:ncol)           & ! WRITE OPERATION - vertically integrated total energy
+      )
+    else
+      ! FV dycore: dycore energy is the same as physics
+      te_ini_dyn(:ncol) = te_ini_phys(:ncol)
+    endif
+
+    ! Set current state to be the same as initial
+    te_cur_phys(:ncol) = te_ini_phys(:ncol)
+    te_cur_dyn (:ncol) = te_ini_dyn (:ncol)
+    tw_cur     (:ncol) = tw_ini     (:ncol)
+
+    ! Zero out current energy unbalances count
+    count = 0
+
+    ! Zero out cumulative boundary fluxes
+    tend_te_tnd(:ncol) = 0._kind_phys
+    tend_tw_tnd(:ncol) = 0._kind_phys
+
+    ! If first timestep, initialize value of teout
+    if(is_first_timestep) then
+      teout(:ncol) = te_ini_dyn(:ncol)
+    endif
+
+  end subroutine check_energy_chng_timestep_init
+
+
+  ! Check that energy and water change matches the boundary fluxes.
+!> \section arg_table_check_energy_chng_run Argument Table
+!! \htmlinclude arg_table_check_energy_chng_run.html
+  subroutine check_energy_chng_run( &
+       ncol, pver, pcnst, &
+       q, pdel, &
+       u, v, T, &
+       pintdry, phis, zm, &
+       cp_phys, &              ! cpairv generally, cpair fixed size for subcolumns code
+       cp_or_cv_dycore, &
+       scaling_dycore, &       ! From check_energy_scaling to work around subcolumns code
+       te_cur_phys, te_cur_dyn, &
+       tw_cur, &
+       tend_te_tnd, tend_tw_tnd, &
+       temp_ini, z_ini, &
+       count, ztodt, &
+       latice, latvap, &
+       energy_formula_physics, energy_formula_dycore, &
+       name, flx_vap, flx_cnd, flx_ice, flx_sen, &
+       errmsg, errflg)
+
+    ! Dependency for hydrostatic energy calculation (physics and dycore formulas)
+    use cam_thermo,         only: get_hydrostatic_energy
+
+    ! Dependency for energy formula used by physics and dynamical cores
+    use cam_thermo_formula, only: ENERGY_FORMULA_DYCORE_FV, ENERGY_FORMULA_DYCORE_SE, ENERGY_FORMULA_DYCORE_MPAS
+
+    ! FIXME hplin: for DEBUG only
+    use cam_logfile,        only: iulog
```

Comment:
You can pass in `iulog` directly as a subroutine argument with the standard name `log_output_unit`.

Reply:
Thanks, fixed!